File: songs/erlang/t09_flatten.erl
-module(t09_flatten).
-import(my_songs, [songs/0]).
-import(lists, [append/1]).
-export([show/0]).

-record(song,  { title, tags = [] }).

show() ->
  Songs = songs(),
  Tags = [ Head#song.tags || Head <- Songs ],
  io:fwrite("~60p~n", [append(Tags)]).

File: songs/scala/T09Flatten.scala
import mysongs._
import mysongs.MySongs._

object T09 extends App {
  def songsFlatten(songs: List[Song])
    = for (
        song <- songs if song.tags != None;
        tag <- song.tags.get
      ) yield tag

  println(songsFlatten(songs))
}
